src/App.js

```diff
@@ -1,25 +1,64 @@
-import logo from './logo.svg';
+// src/App.js
 import './App.css';
+import foods from "./foods.json";
+import { useState } from 'react';
+import FoodBox from './components/FoodBox';
+import AddFoodForm from './components/AddFoodForm';
+import Searchbar from './components/Searchbar';
+
+
 
 function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+  const [foodList, setFoodList] = useState(foods);
+
+  const addFood = (food) => {
+    const foodListCopy = [...foodList, food]
+    setFoodList(foodListCopy)
+  }
+
+  const searchFoods = (query) => {
+    const filteredFoods = foods.filter((food) =>
+    food.name.toLowerCase().includes(query.toLowerCase())
+    );
+    setFoodList(filteredFoods);
+  };
+
+  const deleteFoods = (foods) => {
+    const deletedFoods = foods.filter((food) => {
+      return food.name !== foods;
+    });
+    setFoodList(deletedFoods);
```

Comment:
You're `deleteFoods()` function is not working because you're filtering `foods`(json file) instead of filtering `foodList`(variable):
```
  const deleteFoods = (foods) => {
    const deletedFoods = foodList.filter((food) => {
      return food.name !== foods;
    });
    setFoodList(deletedFoods);
  };
```